tests/lib/rules/no-restricted-imports.js

```diff
@@ -230,6 +230,27 @@ ruleTester.run("no-restricted-imports", rule, {
                 }]
             }]
         },
+        {
+            code: "import { AllowedObject } from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "foo",
+                    allowImportNames: ["AllowedObject"],
+                    message: "Please import anything except 'AllowedObject' from /bar/ instead."
+                }]
+            }]
+        },
+        {
+            code: "import { AllowedObject } from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "foo",
+                    importNames: ["DisallowedObject"],
+                    allowImportNames: ["AllowedObject"],
+                    message: "Please import 'DisallowedObject' from /bar/ instead."
+                }]
+            }]
+        },
```

Comment:
This configuration would be confusing. `DisallowedObject` is disallowed, `AllowedObject` is allowed, but what about other names, e.g., `import { AnotherObject } from "foo";`? If they're also disallowed, then `importNames` seems redundant and it might be better to disallow specifying both `importNames` and `allowImportNames` in the same object.

Reply:
@mdjermanovic  Sure, how would you disallow using both?

Is there are a common approach for this, or should I look into it?

Reply:
I think adding the following to the subschemas that define these objects would work:

```
not: { required: ["importNames", "allowImportNames"] }
```